### PR TITLE
fix: change the annotation from `scheduler-simulator/*` to `kube-scheduler-simulator.sigs.k8s.io/*`

### LINKED ISSUE #386: Move the top level of our annotations to `scheduler-simulator.sigs.k8s.io`
https://github.com/kubernetes/website/pull/47104#discussion_r1833180455

Suggestion from @sftim.

/assign
/kind feature

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area simulator
/area web

/kind feature

#### What this PR does / why we need it:

change the annotation from scheduler-simulator/* to kube-scheduler-simulator.sigs.k8s.io/* 

This is a breaking change which we need to mention at the release note.

#### Which issue(s) this PR fixes:

Fixes #386

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

**Comment by sanposhiho** [comment]
@utam0k 
Sorry for pinging again, can you take a look at this one too when have time?

File: README.md [hunk 1]
```diff
@@ -59,26 +59,26 @@ apiVersion: v1
 metadata:
   name: hoge-pod
```
Comment (utam0k) [inline README.md:60]: BTW: we need to update this name. `hoge` is for Japanese people 😆

Comment (sanposhiho) [reply README.md:60]: Whoa, I didn't know that.

**Pull request merged**